omniscidb/QueryBuilder/QueryBuilder.h

```diff
@@ -0,0 +1,656 @@
+/**
+ * Copyright (C) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "IR/Exception.h"
+#include "IR/Node.h"
+#include "SchemaMgr/SchemaProvider.h"
+#include "Shared/Config.h"
+
+namespace hdk::ir {
+
+class InvalidQueryError : public Error {
+ public:
+  InvalidQueryError() {}
+  InvalidQueryError(std::string desc) : Error(std::move(desc)) {}
+};
+
+class QueryBuilder;
```

Comment:
Does a `QueryBuilder` belong in `hdk::ir` namespace and not just `hdk`?

Reply:
`QueryBuilder` interfaces are purely to manipulate IR, so I guess this is the right namespace. I'm not sure about these two-level namespaces in the first place, but apparently, we already have class names intersections (e.g. we have two `ColumnRef` classes)